### ISSUE TITLE
XADCdemo fails timing

### ISSUE THREAD

**Issue by DiegoRosales** [Digilent/Basys3#1]
After running the tcl script and generating the bitfile, the timing result shows that that there are timing violations related to the digx assignments.

```
            decimal = decimal * 250000;
            decimal = decimal >> 10;


            dig0 = decimal % 10;
            decimal = decimal / 10;

            dig1 = decimal % 10;
            decimal = decimal / 10;

            dig2 = decimal % 10;
            decimal = decimal / 10;

            dig3 = decimal % 10;
            decimal = decimal / 10;

            dig4 = decimal % 10;
            decimal = decimal / 10;

            dig5 = decimal % 10;
            decimal = decimal / 10; 

            dig6 = decimal % 10;
            decimal = decimal / 10; 
```

![image](https://cloud.githubusercontent.com/assets/6503118/14221644/923dbffc-f825-11e5-95a0-1a5c717507f1.png)

A better approach to this may be to use non-blocking assignments and have dedicated logic for each digit.
Other solution may be to pipeline the assignment process.

Note that I'm using Vivaod 2015.2, so this results may vary for newer versions of Vivado. Also, better timing results are obtained by changing the Syn and PAR directives

**Comment by SamKLowe** [comment]
Non-blocking may cause some problems since the code is repeatedly accessing the decimal register. I think the computation would not work as designed. I think pipe-lining would work well for this design since the display digits do not need to be updated as fast as new data is rolling out of the XADC. However, I really think the code needs to ditch the / 10 operations. After surfing the web for awhile, I think a method like the one found at the link below would work a lot better.

http://www.eng.utah.edu/~nmcdonal/Tutorials/BCDTutorial/BCDConversion.html

It's awesome to open up these conversations to improve designs!
